backend/lib/endpoints/feedback.js

```diff
@@ -0,0 +1,42 @@
+const httpErrors = require("http-errors");
+const { createFeedbackSchema } = require("./schema/feedback");
+
+/*
+ * /api/feedback
+ */
+async function routes(app) {
+  const Feedback = app.mongo.model("Feedback");
+
+  app.post("/", { schema: createFeedbackSchema }, async (req) => {
+    const { userId } = req.body;
+    let ip = "undefined";
+    if (userId) {
+      const userFeedback = await Feedback.find({ userId });
+      if (userFeedback[0]) {
+        return new httpErrors.Conflict("User already submitted feedback");
+      }
+    }
+
+    if (req.ip !== null && typeof req.ip !== "undefined") {
+      ip = req.ip;
```

Comment:
From a security standpoint, should we double check what `req.ip` is before we save it to our db? Also, what if the IP address is from a public device accessible by anyone?

Reply:
Good point @anniegiang, maybe we can validate that on this on the schema request with `.format`? It seems that we can have [IPV4 and IPV6 formats](https://github.com/fastify/fluent-schema/blob/master/docs/API.md),

Reply:
What kind of checks do you imagine having here? I am just unsure as to what would be best, because it is coming directly from fastify and I don't know what potential security concerns would be present.

Reply:
@Naraujo13 
Adding a validation for IP addresses using `.format` sounds like a good first step. 

Allowing at most 1 submission for a logged-in user sounds reasonable. 

Regarding public devices - other than validating the IP address, we also need to make sure the user is unique. Perhaps we could generate a unique id for anonymous/not-logged users?

Reply:
@DanielAlexander84 

I was imagining what @Naraujo13 pointed out - validating the format of the address and to at least make sure what's coming in is not malicious. 

Another security concern I just discovered is [IP spoofing](https://en.wikipedia.org/wiki/IP_address_spoofing), which I realize may be out of scope for a simple feedback form.

Reply:
yeah I think it's a great idea to use that IP validation, we can use this for many things coming in the request and I believe we should.
The current version of JSON schema has a lot of validation already built-in it seems, such as e-mail addreses and such, so we should definitely leverage that where we can, and we can also add custom regexes etc. to be more specific instead of just requiring "string:". 
We should also try to specify in the schema where null is allowed and where ist isn't.